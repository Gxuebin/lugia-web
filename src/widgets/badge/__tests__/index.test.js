//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Wrapper from '../demo';

describe(' Badge', () => {
  it('css', () => {
    const Target = <Wrapper />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });
});
