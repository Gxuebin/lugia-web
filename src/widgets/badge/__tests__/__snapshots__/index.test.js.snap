// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge css 1`] = `
.c4 {
  font-size: 1rem;
  box-sizing: border-box;
  position: absolute;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: 10;
  background: #f22735;
  color: #ffffff;
  right: 0em;
  top: -0.8em;
  white-space: nowrap;
  text-align: center;
  overflow: hidden;
  padding: 0 0.2em;
  color: white;
  display: inline-block;
  height: 1.4em;
  width: 1.4em;
  border-radius: 0.8em;
  line-height: 1.4em;
  text-align: center;
  padding: 0 0.4em;
  font-weight: normal;
  white-space: nowrap;
  -webkit-box-shadow: 0 0 0 0.1em #fff;
  box-shadow: 0 0 0 0.1em #fff;
}

.c7 {
  font-size: 1rem;
  box-sizing: border-box;
  position: absolute;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: 10;
  background: #f22735;
  color: #ffffff;
  right: 0em;
  top: -0.8em;
  white-space: nowrap;
  text-align: center;
  overflow: hidden;
  padding: 0 0.2em;
  color: white;
  display: inline-block;
  height: 1.4em;
  width: 2em;
  border-radius: 0.8em;
  line-height: 1.4em;
  text-align: center;
  padding: 0 0.4em;
  font-weight: normal;
  white-space: nowrap;
  -webkit-box-shadow: 0 0 0 0.1em #fff;
  box-shadow: 0 0 0 0.1em #fff;
}

.c11 {
  font-size: 1rem;
  box-sizing: border-box;
  position: absolute;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: 10;
  background: orange;
  color: black;
  right: 0em;
  top: -0.8em;
  white-space: nowrap;
  text-align: center;
  overflow: hidden;
  padding: 0 0.2em;
  color: white;
  display: inline-block;
  height: 1.4em;
  width: 1.4em;
  border-radius: 0.8em;
  line-height: 1.4em;
  text-align: center;
  padding: 0 0.4em;
  font-weight: normal;
  white-space: nowrap;
  -webkit-box-shadow: 0 0 0 0.1em #fff;
  box-shadow: 0 0 0 0.1em #fff;
}

.c5 {
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  white-space: nowrap;
  text-align: center;
  -webkit-transform: translateY(-400%);
  -ms-transform: translateY(-400%);
  transform: translateY(-400%);
  height: 1.4em;
  display: inline-block;
}

.c8 {
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  white-space: nowrap;
  text-align: center;
  -webkit-transform: translateY(-900%);
  -ms-transform: translateY(-900%);
  transform: translateY(-900%);
  height: 1.4em;
  display: inline-block;
}

.c9 {
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  white-space: nowrap;
  text-align: center;
  -webkit-transform: translateY(-808%);
  -ms-transform: translateY(-808%);
  transform: translateY(-808%);
  height: 1.4em;
  display: inline-block;
}

.c12 {
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  white-space: nowrap;
  text-align: center;
  -webkit-transform: translateY(0%);
  -ms-transform: translateY(0%);
  transform: translateY(0%);
  height: 1.4em;
  display: inline-block;
}

.c6 {
  height: 1.4em;
  text-align: center;
  box-sizing: border-box;
  margin: 0;
}

.c10 {
  font-size: 1rem;
  box-sizing: border-box;
  position: absolute;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: 10;
  background: #f22735;
  color: #ffffff;
  right: 0em;
  top: -0.5em;
  height: 1em;
  width: 1em;
  border-radius: 100%;
  z-index: 10;
  box-shadow: 0 0 0 0.08333333333333334em #fff;
}

.c13 {
  font-size: 1rem;
  box-sizing: border-box;
  position: absolute;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: 10;
  background: orange;
  color: black;
  right: 0em;
  top: -0.5em;
  height: 1em;
  width: 1em;
  border-radius: 100%;
  z-index: 10;
  box-shadow: 0 0 0 0.08333333333333334em #fff;
}

.c14 {
  font-size: 1rem;
  box-sizing: border-box;
  position: absolute;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: 10;
  background: green;
  color: #ffffff;
  right: 0em;
  top: -0.5em;
  height: 1em;
  width: 1em;
  border-radius: 100%;
  z-index: 10;
  box-shadow: 0 0 0 0.08333333333333334em #fff;
}

.c15 {
  font-size: 1rem;
  box-sizing: border-box;
  position: absolute;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: 10;
  background: purple;
  color: #ffffff;
  right: 0em;
  top: -0.5em;
  height: 1em;
  width: 1em;
  border-radius: 100%;
  z-index: 10;
  box-shadow: 0 0 0 0.08333333333333334em #fff;
}

.c16 {
  font-size: 1rem;
  box-sizing: border-box;
  position: absolute;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: 10;
  background: yellow;
  color: #ffffff;
  right: 0em;
  top: -0.5em;
  height: 1em;
  width: 1em;
  border-radius: 100%;
  z-index: 10;
  box-shadow: 0 0 0 0.08333333333333334em #fff;
}

.c17 {
  font-size: 1rem;
  box-sizing: border-box;
  position: absolute;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  z-index: 10;
  background: blue;
  color: #ffffff;
  right: 0em;
  top: -0.5em;
  height: 1em;
  width: 1em;
  border-radius: 100%;
  z-index: 10;
  box-shadow: 0 0 0 0.08333333333333334em #fff;
}

.c2 {
  background: transparent;
  box-sizing: border-box;
  position: relative;
  display: inline-block;
}

.c0 {
  position: relative;
  margin-left: 0;
  margin-right: 0;
  height: auto;
  zoom: 1;
  display: block;
  box-sizing: border-box;
  margin: 4.166666666666667em;
}

.c0:after {
  content: "";
  clear: both;
  display: block;
  width: 0;
  height: 0;
  visibility: hidden;
}

.c0::before {
  content: "";
  display: table;
}

.c1 {
  display: block;
  box-sizing: border-box;
  float: left;
  position: relative;
  width: 16.666666666666668%;
  margin-left: 0%;
}

.c18 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c3 {
  width: 40px;
  height: 40px;
  background: #ccc;
  margin-left: 10px;
}

<div>
  <span>
    <div
      className="c0"
      type="default"
    >
      <div
        className="c1"
        span={4}
      >
        <span
          className="c2"
        >
          <div
            className="c3"
          />
          <span>
            <div
              className=""
            >
              <sup
                className="c4"
                overflow={false}
              >
                <span
                  className="c5"
                  y={-400}
                >
                  <p
                    className="c6"
                  >
                    0
                  </p>
                  <p
                    className="c6"
                  >
                    1
                  </p>
                  <p
                    className="c6"
                  >
                    2
                  </p>
                  <p
                    className="c6"
                  >
                    3
                  </p>
                  <p
                    className="c6"
                  >
                    4
                  </p>
                  <p
                    className="c6"
                  >
                    5
                  </p>
                  <p
                    className="c6"
                  >
                    6
                  </p>
                  <p
                    className="c6"
                  >
                    7
                  </p>
                  <p
                    className="c6"
                  >
                    8
                  </p>
                  <p
                    className="c6"
                  >
                    9
                  </p>
                </span>
              </sup>
            </div>
          </span>
        </span>
      </div>
      <div
        className="c1"
        span={4}
      >
        <span
          className="c2"
        >
          <div
            className="c3"
          />
          <span>
            <div
              className=""
            >
              <sup
                className="c7"
                overflow={false}
              >
                <span
                  className="c8"
                  y={-900}
                >
                  <p
                    className="c6"
                  >
                    0
                  </p>
                  <p
                    className="c6"
                  >
                    1
                  </p>
                  <p
                    className="c6"
                  >
                    2
                  </p>
                  <p
                    className="c6"
                  >
                    3
                  </p>
                  <p
                    className="c6"
                  >
                    4
                  </p>
                  <p
                    className="c6"
                  >
                    5
                  </p>
                  <p
                    className="c6"
                  >
                    6
                  </p>
                  <p
                    className="c6"
                  >
                    7
                  </p>
                  <p
                    className="c6"
                  >
                    8
                  </p>
                  <p
                    className="c6"
                  >
                    9
                  </p>
                </span>
                <span
                  className="c8"
                  y={-900}
                >
                  <p
                    className="c6"
                  >
                    0
                  </p>
                  <p
                    className="c6"
                  >
                    1
                  </p>
                  <p
                    className="c6"
                  >
                    2
                  </p>
                  <p
                    className="c6"
                  >
                    3
                  </p>
                  <p
                    className="c6"
                  >
                    4
                  </p>
                  <p
                    className="c6"
                  >
                    5
                  </p>
                  <p
                    className="c6"
                  >
                    6
                  </p>
                  <p
                    className="c6"
                  >
                    7
                  </p>
                  <p
                    className="c6"
                  >
                    8
                  </p>
                  <p
                    className="c6"
                  >
                    9
                  </p>
                </span>
                <span
                  className="c9"
                  y={-808}
                >
                  +
                </span>
              </sup>
            </div>
          </span>
        </span>
      </div>
      <div
        className="c1"
        span={4}
      >
        <span
          className="c2"
        >
          <div
            className="c3"
          />
          <sup
            className="c10"
          />
        </span>
      </div>
    </div>
    <span>
      <div
        className="c0"
        type="default"
      >
        <div
          className="c1"
          span={4}
        >
          <span
            className="c2"
          >
            <div
              className="c3"
            />
          </span>
        </div>
        <div
          className="c1"
          span={4}
        >
          <span
            className="c2"
          >
            <div
              className="c3"
            />
            <span>
              <div
                className=""
              >
                <sup
                  className="c11"
                  overflow={false}
                >
                  <span
                    className="c12"
                    y={-0}
                  >
                    <p
                      className="c6"
                    >
                      0
                    </p>
                    <p
                      className="c6"
                    >
                      1
                    </p>
                    <p
                      className="c6"
                    >
                      2
                    </p>
                    <p
                      className="c6"
                    >
                      3
                    </p>
                    <p
                      className="c6"
                    >
                      4
                    </p>
                    <p
                      className="c6"
                    >
                      5
                    </p>
                    <p
                      className="c6"
                    >
                      6
                    </p>
                    <p
                      className="c6"
                    >
                      7
                    </p>
                    <p
                      className="c6"
                    >
                      8
                    </p>
                    <p
                      className="c6"
                    >
                      9
                    </p>
                  </span>
                </sup>
              </div>
            </span>
          </span>
        </div>
        <div
          className="c1"
          span={4}
        >
          <span
            className="c2"
          >
            <div
              className="c3"
            />
            <span>
              <div
                className=""
              >
                <sup
                  className="c11"
                  overflow={false}
                >
                  <span
                    className="c12"
                    y={-0}
                  >
                    <p
                      className="c6"
                    >
                      0
                    </p>
                    <p
                      className="c6"
                    >
                      1
                    </p>
                    <p
                      className="c6"
                    >
                      2
                    </p>
                    <p
                      className="c6"
                    >
                      3
                    </p>
                    <p
                      className="c6"
                    >
                      4
                    </p>
                    <p
                      className="c6"
                    >
                      5
                    </p>
                    <p
                      className="c6"
                    >
                      6
                    </p>
                    <p
                      className="c6"
                    >
                      7
                    </p>
                    <p
                      className="c6"
                    >
                      8
                    </p>
                    <p
                      className="c6"
                    >
                      9
                    </p>
                  </span>
                </sup>
              </div>
            </span>
          </span>
        </div>
        <div
          className="c1"
          span={4}
        >
          <a
            href="www.baidu.com"
          >
            <span
              className="c2"
            >
              <div
                className="c3"
              />
              <sup
                className="c13"
              />
            </span>
          </a>
        </div>
      </div>
    </span>
    <div
      className="c0"
      type="default"
    >
      <div
        className="c1"
        span={4}
      >
        <span>
          <span
            className="c2"
          >
            <div
              className="c3"
            />
            <sup
              className="c14"
            />
          </span>
        </span>
         
      </div>
      <div
        className="c1"
        span={4}
      >
        <span>
          <span
            className="c2"
          >
            <div
              className="c3"
            />
            <sup
              className="c15"
            />
          </span>
        </span>
         
      </div>
      <div
        className="c1"
        span={4}
      >
        <span>
          <span
            className="c2"
          >
            <div
              className="c3"
            />
            <sup
              className="c16"
            />
          </span>
        </span>
         
      </div>
      <div
        className="c1"
        span={4}
      >
        <span>
          <span
            className="c2"
          >
            <div
              className="c3"
            />
            <sup
              className="c17"
            />
          </span>
        </span>
      </div>
    </div>
    <div
      className="c0"
      type="default"
    >
      <div
        className="c1"
        span={4}
      >
        <span
          className="c2"
        >
          <div
            className="c3"
          />
          <span>
            <div
              className=""
            >
              <sup
                className="c4"
                overflow={false}
              >
                <span
                  className="c12"
                  y={-0}
                >
                  <p
                    className="c6"
                  >
                    0
                  </p>
                  <p
                    className="c6"
                  >
                    1
                  </p>
                  <p
                    className="c6"
                  >
                    2
                  </p>
                  <p
                    className="c6"
                  >
                    3
                  </p>
                  <p
                    className="c6"
                  >
                    4
                  </p>
                  <p
                    className="c6"
                  >
                    5
                  </p>
                  <p
                    className="c6"
                  >
                    6
                  </p>
                  <p
                    className="c6"
                  >
                    7
                  </p>
                  <p
                    className="c6"
                  >
                    8
                  </p>
                  <p
                    className="c6"
                  >
                    9
                  </p>
                </span>
              </sup>
            </div>
          </span>
        </span>
        <i
          className="lugia-icon-reminder_plus_square_o  c18"
          onClick={[Function]}
        />
        <i
          className="lugia-icon-reminder_minus_square_o  c18"
          onClick={[Function]}
        />
      </div>
    </div>
  </span>
</div>
`;
